### ISSUE TITLE
Does the gumbel_sample function have another version?

### ISSUE THREAD

**Issue by 2502912390** [avishkarsaha/diff-graph-gen#2]
Hello, thank you for your work.
The gumbel_sample function seems to merely add noise to the logits, without performing softmax sampling.

def gumbel_sample(logits, noise_sample):
    """Draw a sample from the Gumbel-Softmax distribution
    Args:
        logits: input log probabilities of size [N, N]
    """
    assert logits.shape == noise_sample.shape

    zero_self_loops = 1 - torch.eye(n=logits.shape[0])

    if torch.cuda.is_available():
        zero_self_loops = zero_self_loops.cuda()

    noise = noise_sample * zero_self_loops
    y = logits + noise_sample
    # y = logits + noise #？？？
    return y

gumbel_sample(logits, temperature, self_loops_noise)
